### PR TITLE
Add bolt tasks

### PR DESCRIPTION
**Pull request by treydock**
# Pull Request Checklist

## Description

Add bolt tasks to manage sensuctl events and silencings. These will replace sensu_event and sensu_silenced types removed in #1141 

## Related Issue

Part of #1147 

## Motivation and Context

Events and silencings are ad-hoc tasks so best to move into bolt tasks.

### CODE REVIEW

**Comment by treydock** [comment]
Added bolt tasks to install sensu go agent on Linux and Windows.

File: tasks/silenced.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
+#!/opt/puppetlabs/puppet/bin/ruby
+require 'json'
+require 'open3'
+require 'puppet'
+require 'tempfile'
+
+def sensuctl_create(name, spec, namespace)
```
Comment (ghoneycutt) [inline tasks/silenced.rb]: Does bolt have puppet string support? How is it meant to be documented?

Comment (treydock) [reply tasks/silenced.rb]: I believe the JSON metadata generates documentation on Forge. https://forge.puppet.com/puppetlabs/exec/tasks

The docs also mention tasks getting automatic docs from strings using JSON metadata: https://puppet.com/docs/puppet/5.5/puppet_strings_style.html#puppet-tasks-and-plans

**Comment by treydock** [comment]
Removed WIP. Added all tasks mentioned in #1147

File: tests/test_bolt_win.ps1 [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#$env:PATH += ";C:\Program Files\sensu\sensu-agent\bin"
```
Comment (ghoneycutt) [inline tests/test_bolt_win.ps1]: why the commented out code?

File: tasks/silenced.rb [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    }
+    result = sensuctl_create(name, spec, namespace)
+  end
+  puts result.to_json
+  exit 0
+rescue Puppet::Error => e
+  puts({ status: 'failure', error: e.message }.to_json)
+  exit 1
+end
+
```
Comment (ghoneycutt) [inline tasks/silenced.rb]: style: no empty lines at the bottom of file

File: tasks/event.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  when 'delete'
+    result = delete(entity, check, namespace)
+  end
+  puts result.to_json
+  exit 0
+rescue Puppet::Error => e
+  puts({ status: 'failure', error: e.message }.to_json)
+  exit 1
+end
+
```
Comment (ghoneycutt) [inline tasks/event.rb]: style: no trailing empty lines

File: tests/test_bolt_win.ps1 (hunk 2 above)
Comment (treydock) [reply tests/test_bolt_win.ps1]: Was left over while debugging Windows issues. I removed.

File: tasks/silenced.rb (hunk 3 above)
Comment (treydock) [reply tasks/silenced.rb]: Removed

File: tasks/event.rb (hunk 4 above)
Comment (treydock) [reply tasks/event.rb]: Removed

File: tasks/install_agent_linux.rb [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+  ret = {status: "install agent successful"}
+  ret['output'] = return_output if output
+  puts(ret.to_json)
+  exit 0
+rescue Puppet::Error => e
+  puts({ status: 'failure', error: e.message }.to_json)
+  exit 1
+end
+
```
Comment (ghoneycutt) [inline tasks/install_agent_linux.rb]: no empty trailing lines

File: tasks/install_agent_linux.rb [hunk 6]
```diff
@@ -0,0 +1,61 @@
+#!/opt/puppetlabs/puppet/bin/ruby
```
Comment (ghoneycutt) [inline tasks/install_agent_linux.rb]: This file is pretty big to not have any docs. Please add some docs to describe what it is doing and perhaps some inline ones as you see fit.

File: tasks/event.rb [hunk 7]
```diff
@@ -0,0 +1,36 @@
+#!/opt/puppetlabs/puppet/bin/ruby
```
Comment (ghoneycutt) [inline tasks/event.rb]: This file is pretty big to not have any docs. Please add some docs to describe what it is doing and perhaps some inline ones as you see fit.

File: tasks/check_execute.rb [hunk 8]
```diff
@@ -0,0 +1,32 @@
+#!/opt/puppetlabs/puppet/bin/ruby
```
Comment (ghoneycutt) [inline tasks/check_execute.rb]: Add docs

File: tasks/agent_event.rb [hunk 9]
```diff
@@ -0,0 +1,37 @@
+#!/opt/puppetlabs/puppet/bin/ruby
```
Comment (ghoneycutt) [inline tasks/agent_event.rb]: Add docs

File: spec/spec_helper_acceptance.rb [hunk 10]
```diff
@@ -108,5 +109,16 @@
       create_remote_file(puppetserver, '/etc/puppetlabs/code/environments/production/manifests/site.pp', '')
       on puppetserver, "chmod 0644 /etc/puppetlabs/code/environments/production/manifests/site.pp"
     end
+
+    on setup_nodes, puppet("resource package puppet-bolt ensure=installed")
+    bolt_cfg = <<-EOS
+modulepath: "/etc/puppetlabs/code/modules:/etc/puppetlabs/code/environments/production/modules"
+ssh:
```
Comment (ghoneycutt) [inline spec/spec_helper_acceptance.rb]: Document this is because of bolt

File: spec/spec_helper_acceptance.rb [hunk 11]
```diff
@@ -16,6 +16,7 @@
   c.add_setting :sensu_test_enterprise, default: false
   c.add_setting :sensu_manage_repo, default: true
   c.add_setting :sensu_use_agent, default: false
+  c.add_setting :skip_apply, default: false
```
Comment (ghoneycutt) [inline spec/spec_helper_acceptance.rb]: This file is pretty big to not have any docs. Please add some docs to describe what it is doing and perhaps some inline ones as you see fit.

File: tasks/install_agent_linux.rb (hunk 5 above)
Comment (treydock) [reply tasks/install_agent_linux.rb]: Fixed.

File: tasks/install_agent_linux.rb (hunk 6 above)
Comment (treydock) [reply tasks/install_agent_linux.rb]: I added some inline but the main documentation is handled by Puppet Strings.

File: tasks/event.rb (hunk 7 above)
Comment (treydock) [reply tasks/event.rb]: The main docs are handled by Puppet Strings based on the JSON file, not much going on inside this file that's not described in JSON.

File: tasks/check_execute.rb (hunk 8 above)
Comment (treydock) [reply tasks/check_execute.rb]: What docs would be useful that aren't handled by the JSON task definition? The Ruby code is pretty basic.

File: tasks/agent_event.rb (hunk 9 above)
Comment (treydock) [reply tasks/agent_event.rb]: Just like other simpler tasks: What docs would be useful that aren't handled by the JSON task definition? The Ruby code is pretty basic.

File: spec/spec_helper_acceptance.rb (hunk 10 above)
Comment (treydock) [reply spec/spec_helper_acceptance.rb]: Comment added at top, rest is pretty obvious based on variable names and having "bolt" in commands".

File: spec/spec_helper_acceptance.rb (hunk 11 above)
Comment (treydock) [reply spec/spec_helper_acceptance.rb]: This particular setting isn't used, I've removed it.

**Pull request merged**